frontend/src/api/index.ts

```diff
@@ -31,3 +31,6 @@ export * from './errorUtils';
 
 // User access review hook
 export * from './useAccessReview';
+
+// API Models
+export * from './models';
```

Comment:
we didn’t expose the models to import.
cc @lucferbux

Reply:
This is a valid point, tagging here @andrewballantyne, @christianvogt and @alexcreasy in case they have some thoughts.
This workaround is located in the `services` and `utilities` folders, outside the `api` scope, but we need models. I cannot recall if we didn't export that for a reason or if it's safe to export models too.

Reply:
I'm going to need context.

Why do you need to touch anything in `services`? More over, why do you need models outside of the API? Sounds like you're breaking the intentional encapsulation of our K8s models. You need them outside of the API?

Reply:
I can reply that, this is not pass through api, we are touching requests to the backend, which were placed in the `service` folder.
To keep that structure, when I applied the workaround for Custom Serving Runtime I placed the calls in the `service` folder.
So I see three alternatives here:

- Move the utility function into api and then import that into `services`
- Move all the logic of the backend to the `api` foler, which i discourage since I always thought it was targeted for the pass-trough/pipelines
- Export models and get this suggested layout

Reply:
Maybe @christianvogt has a suggestion here -- my recommendation would be to deprecate the entire folder of services and if we need custom api end points to craft them all in `/api` -- maybe `/api/internal` or something

We should absolutely not break the flow of importing of API models just because of convenience. `services` has a lot of the "bad" code -- I'd like not to mix the new (and current plan) with the old (jupyter tile like flows)

Reply:
TBH I'll need more background on the history of services and the path forward to comment fully.

But it makes sense not to export the models if the intent is that all functions that work with models should be contained in the api folder and services is deprecated as an outlier. That being said if services is API like and deprecated, why not allow direct imports from the _services_ directory as well?

Aside from that, the new api `addTypesToK8sListedResources` is only needed when not using the SDK (which already does this same operation). We could also make a one off hard coded value of the `Template` kind in the service itself.

The function could also be simplified to just accept `kind` in this case:
`addTypesToK8sListedResources(response.data, 'Template')`
We only need the value of `kind` since `apiVersion` is obtainable from `response.data.apiVersion`.

For the sake of this PR, i suggest we do one of the above but not export models entirely. If further refactoring of the services dir is needed, a tech debt item with more details should be created.